tsdb/index/postings_test.go

```diff
@@ -718,6 +719,847 @@ func TestBigEndian(t *testing.T) {
 	})
 }
 
+func TestPrefixCompressedPostings(t *testing.T) {
+	num := 1000
+	// mock a list as postings
+	ls := make([]uint64, num)
+	ls[0] = 2
+	for i := 1; i < num; i++ {
+		ls[i] = ls[i-1] + uint64(rand.Int31n(25)) + 2
+	}
+
+	buf := encoding.Encbuf{}
+	writePrefixCompressedPostings(&buf, ls)
+
+	t.Run("Iteration", func(t *testing.T) {
+		rbp := newPrefixCompressedPostings(buf.Get())
+		for i := 0; i < num; i++ {
+			testutil.Assert(t, rbp.Next() == true, "")
+			testutil.Equals(t, uint64(ls[i]), rbp.At())
+		}
+
+		testutil.Assert(t, rbp.Next() == false, "")
+		testutil.Assert(t, rbp.Err() == nil, "")
+	})
+
+	t.Run("Seek", func(t *testing.T) {
+		table := []struct {
+			seek  uint64
+			val   uint64
+			found bool
+		}{
+			{
+				ls[0] - 1, ls[0], true,
+			},
+			{
+				ls[4], ls[4], true,
+			},
+			{
+				ls[500] - 1, ls[500], true,
+			},
+			{
+				ls[600] + 1, ls[601], true,
+			},
+			{
+				ls[600] + 1, ls[601], true,
+			},
+			{
+				ls[600] + 1, ls[601], true,
+			},
+			{
+				ls[0], ls[601], true,
+			},
+			{
+				ls[600], ls[601], true,
+			},
+			{
+				ls[999], ls[999], true,
+			},
+			{
+				ls[999] + 10, ls[999], false,
+			},
+		}
+
+		rbp := newPrefixCompressedPostings(buf.Get())
+
+		for _, v := range table {
+			testutil.Equals(t, v.found, rbp.Seek(uint64(v.seek)))
+			testutil.Equals(t, uint64(v.val), rbp.At())
+			testutil.Assert(t, rbp.Err() == nil, "")
+		}
+	})
+}
+
+func BenchmarkPostingsOneBlock(b *testing.B) {
+	num := 1000
+	ls1 := make([]uint32, num) // Block with key 0.
+	ls2 := make([]uint32, num) // Block with key > 0.
+	ls1[0] = 2
+	ls2[0] = 655360
+	for i := 1; i < num; i++ {
+		ls1[i] = ls1[i-1] + uint32(rand.Int31n(10)) + 2
+		ls2[i] = ls2[i-1] + uint32(rand.Int31n(10)) + 2
+	}
+
+	// bigEndianPostings for ls1.
+	bufBE1 := make([]byte, num*4)
+	for i := 0; i < num; i++ {
+		b := bufBE1[i*4 : i*4+4]
+		binary.BigEndian.PutUint32(b, ls1[i])
+	}
+
+	// prefixCompressedPostings for ls1.
+	bufPCP1 := encoding.Encbuf{}
+	temp := make([]uint64, len(ls1))
+	for i, x := range ls1 {
+		temp[i] = uint64(x)
+	}
+	writePrefixCompressedPostings(&bufPCP1, temp)
```

Comment:
using internal functions makes it impossible to run proper benchmarks - new vs old code.
I mean the internal implementation changes quite frequently so this also means refactoring the bench test as well, making it impossible to run the same bench test on the new and the old code.

Can you implement these using only external API so that even when the internal implementation changes we can still run these  without any changes to the bench tests.

This will also allow us to use the automated /funcbench testing project that we have been working on.